tests/lsdb/loaders/dataframe/test_read_dataframe.py

```diff
@@ -0,0 +1,40 @@
+import pandas as pd
+import pytest
+from hipscat.catalog import CatalogType
+
+import lsdb
+
+
+def test_read_catalog_from_dataframe(small_sky_order1_catalog, small_sky_order1_csv):
+    """Tests that we can initialize a catalog from a Pandas Dataframe
+    and that the loaded content is correct."""
+    # Read CSV file for the small sky order 1 catalog
+    catalog = lsdb.read_dataframe(
+        path=small_sky_order1_csv,
+        catalog_name="small_sky_order1",
+        catalog_type=CatalogType.SOURCE
+    )
+    assert isinstance(catalog, lsdb.Catalog)
+
+    # Catalogs have the same information
+    assert catalog.hc_structure.catalog_info == small_sky_order1_catalog.hc_structure.catalog_info
+
+    # Data frames have the same data (column data types may differ)
+    pd.testing.assert_frame_equal(
+        catalog.compute().reset_index(drop=True),
+        small_sky_order1_catalog.compute().reset_index(drop=True),
+        check_column_type=False
+    )
+
+
```

Comment:
Some extra unit tests I would like:
 - check the partitions of the dask df match the partitions in the ddf_partition_map
 - check the partitions of the ddf_partition_map match the pixel_tree and partition_info in the hc_structure
 - check using non default ra + dec column names
 - only allow catalog types that use ra dec partitioning with data points